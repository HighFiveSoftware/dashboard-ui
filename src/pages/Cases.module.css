.main_container {
    max-width: 85%;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    justify-content: space-between;
    /* align-content: space-around; */
}

.charts_container {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}