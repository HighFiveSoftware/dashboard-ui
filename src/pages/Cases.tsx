import { RouteComponentProps } from '@reach/router';
import React from 'react';
import { Container, Grid } from 'semantic-ui-react';
import { DailyCaseChart } from '../components/DailyCaseChart';
import { DailyCaseStatistics } from '../components/DailyCaseStatistics';
import styles from './Cases.module.css';

const vals = [
  {
    name: 'confirmed',
    value: 1000,
    dif: 10
  },
  {
    name: 'recoveries',
    value: 1000,
    dif: 10
  },
  {
    name: 'deaths',
    value: 1000,
    dif: 10
  }
];

export const Cases = (_props: RouteComponentProps) => {
  return (
    <Container className={styles.main_container}>
      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <DailyCaseStatistics title="Worldwide Cases" vals={vals} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <DailyCaseChart title="New Cases" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <DailyCaseChart title="Total Cases" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
