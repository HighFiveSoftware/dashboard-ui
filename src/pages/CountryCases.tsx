import { RouteComponentProps } from '@reach/router';
import { Container, Grid, Message } from 'semantic-ui-react';
import { CaseChart } from '../components/CaseChart';
import { CaseStatistics } from '../components/CaseStatistics';
import styles from './Cases.module.css';
import { capitalize } from '../utils/utils';
import { useCases } from '../hooks/useCases';

interface ICountryCasesProps extends RouteComponentProps {
  country?: string;
}

export const CountryCases = ({ country }: ICountryCasesProps) => {
  const [cases, error, loading] = useCases(country);
  if (loading) return <div>Loading</div>;

  return (
    <Container className={styles.main_container}>
      {error && (
        <Message negative>
          <Message.Header>Error: {error.message}</Message.Header>
        </Message>
      )}

      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <CaseStatistics
              title={`${country ? capitalize(country) : 'Worldwide'} Cases`}
              values={
                cases.length > 0
                  ? [
                      cases[cases.length - 1].confirmedToday,
                      cases[cases.length - 1].deathsToday,
                      cases[cases.length - 1].recoveredToday
                    ]
                  : [0, 0, 0]
              }
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="Total Cases" values={cases} dataKey="today" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="New Cases" values={cases} dataKey="change" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
