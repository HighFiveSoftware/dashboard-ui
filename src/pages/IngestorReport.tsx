import React from 'react';
import { Table } from 'semantic-ui-react';
import { RouteComponentProps } from '@reach/router';
import ingestorReports from '../data/ingestorReports.json';

// eslint-disable-next-line no-unused-vars
export const IngestorReport = (props: RouteComponentProps) => (
  <Table celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>Date</Table.HeaderCell>
        <Table.HeaderCell>Count</Table.HeaderCell>
        <Table.HeaderCell>Time</Table.HeaderCell>
      </Table.Row>
    </Table.Header>

    <Table.Body>
      {ingestorReports.map(
        ({ success, ingestionDate, ingestedCount, ingestionTime }) => (
          <Table.Row positive={success} negative={!success}>
            <Table.Cell>{ingestionDate}</Table.Cell>
            <Table.Cell>{ingestedCount}</Table.Cell>
            <Table.Cell>{ingestionTime} minutes</Table.Cell>
          </Table.Row>
        )
      )}
    </Table.Body>
  </Table>
);
