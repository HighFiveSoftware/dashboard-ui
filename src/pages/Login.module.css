.login_container {
    /* height: 100%; */
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

.login_card {
    max-width: 30%;
}