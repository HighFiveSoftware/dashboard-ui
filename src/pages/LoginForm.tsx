import React, { ChangeEvent, useContext, useState } from 'react';
import {
  Button,
  Form,
  Grid,
  Header,
  InputOnChangeData,
  Message,
  Segment
} from 'semantic-ui-react';
import { RouteComponentProps } from '@reach/router';
import LoginContext from '../context/LoginContext';

interface IFormState {
  email: string;
  password: string;
  error: boolean;
}

const defaultFormState = {
  email: '',
  password: '',
  error: false
};

// eslint-disable-next-line no-unused-vars
export const LoginForm = (props: RouteComponentProps) => {
  const { setLoggedIn } = useContext(LoginContext);

  const [formState, setFormState] = useState<IFormState>(defaultFormState);

  const handleSubmit = () => {
    if (formState.email === 'admin' && formState.password === '123') {
      setFormState(defaultFormState);
      setLoggedIn(true);
    } else {
      setFormState({ ...formState, error: true });
    }
  };

  const handleChange = (
    _e: ChangeEvent<HTMLInputElement>,
    { name, value }: InputOnChangeData
  ) => {
    setFormState({ ...formState, [name]: value });
  };

  return (
    <Grid textAlign="center" style={{ height: '100vh' }} verticalAlign="middle">
      <Grid.Column style={{ maxWidth: 450 }}>
        <Header as="h2" textAlign="center">
          Log-in to your account
        </Header>
        <Form size="large" onSubmit={handleSubmit} error={formState.error}>
          {formState.error && (
            <Message
              error
              header="Login Failed!"
              content="Wrong email or password. Please check again."
            />
          )}
          <Segment stacked>
            <Form.Input
              fluid
              icon="user"
              iconPosition="left"
              placeholder="E-mail address"
              name="email"
              required
              value={formState.email}
              onChange={handleChange}
            />
            <Form.Input
              fluid
              icon="lock"
              iconPosition="left"
              placeholder="Password"
              type="password"
              name="password"
              required
              value={formState.password}
              onChange={handleChange}
            />

            <Button fluid size="large">
              Login
            </Button>
          </Segment>
        </Form>
      </Grid.Column>
    </Grid>
  );
};
