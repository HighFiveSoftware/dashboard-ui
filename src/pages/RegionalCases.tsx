import { RouteComponentProps } from '@reach/router';
import { Grid, Message } from 'semantic-ui-react';
import { CaseChart } from '../components/CaseChart';
import { CaseStatistics } from '../components/CaseStatistics';
import { capitalize } from '../utils/utils';
import { useCases } from '../hooks/useCases';

interface IRegionalCasesProps extends RouteComponentProps {
  region?: string;
}

export const RegionalCases = ({ region }: IRegionalCasesProps) => {
  const [cases, error, loading] = useCases(region);
  if (loading) return <div>Loading</div>;

  return (
    <div>
      {error && (
        <Message negative>
          <Message.Header>Error: {error}</Message.Header>
        </Message>
      )}

      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <CaseStatistics
              title={`${region ? capitalize(region) : 'Worldwide'} Cases`}
              values={
                cases.length > 0
                  ? [
                      cases[cases.length - 1].confirmedToday,
                      cases[cases.length - 1].deathsToday,
                      cases[cases.length - 1].recoveredToday
                    ]
                  : [0, 0, 0]
              }
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="Total Cases" values={cases} dataKey="today" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="New Cases" values={cases} dataKey="change" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </div>
  );
};
