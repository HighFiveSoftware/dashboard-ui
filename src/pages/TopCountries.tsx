import { RouteComponentProps } from '@reach/router';
import { Card, Message } from 'semantic-ui-react';
import { TopCountryTable } from '../components/TopCountryTable';
import { useTopCountries } from '../hooks/useTopCountries';

export const TopCountries = (_props: RouteComponentProps) => {
  const [topCountries, error, loading] = useTopCountries();

  if (loading) return <div>Loading</div>;

  return (
    <div>
      {error && (
        <Message negative>
          <Message.Header>Error: {error.message}</Message.Header>
        </Message>
      )}

      <Card fluid>
        <Card.Content>
          <Card.Header>Top Countries</Card.Header>
        </Card.Content>
        <TopCountryTable topCases={topCountries} />
      </Card>
    </div>
  );
};
