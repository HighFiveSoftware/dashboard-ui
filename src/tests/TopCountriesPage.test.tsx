/* eslint-env jest */
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { screen, waitForElementToBeRemoved } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import topCountryData from './data/topCountryData.json';
import { TopCountries } from '../pages/TopCountries';
import { renderWithRouterWrapper } from './utils';
import { API_URL } from '../constants';

const server = setupServer(
  rest.get(`${API_URL}/cases/topCountries`, (req, res, ctx) => {
    return res(ctx.json(topCountryData));
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('<TopCountries />', () => {
  test('top countries page renders as expected', async () => {
    const component = renderWithRouterWrapper(
      <TopCountries path="/topCountries" />,
      {
        route: '/topCountries'
      }
    );

    await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));

    expect(component.asFragment()).toMatchSnapshot();
  });
});
