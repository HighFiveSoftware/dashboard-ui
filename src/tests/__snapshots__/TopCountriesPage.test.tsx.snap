// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopCountries /> top countries page renders as expected 1`] = `
<DocumentFragment>
  <div
    style="outline: none;"
    tabindex="-1"
  >
    <div>
      <div
        class="ui fluid card"
      >
        <div
          class="content"
        >
          <div
            class="header"
          >
            Top Countries
          </div>
        </div>
        <table
          class="ui table"
          role="table"
        >
          <thead
            class=""
          >
            <tr
              class=""
              role="row"
            >
              <th
                class=""
                colspan="1"
                role="columnheader"
              >
                Country
              </th>
              <th
                class=""
                colspan="1"
                role="columnheader"
              >
                Cases
              </th>
              <th
                class=""
                colspan="1"
                role="columnheader"
              >
                Deaths
              </th>
              <th
                class=""
                colspan="1"
                role="columnheader"
              >
                Recoveries
              </th>
            </tr>
          </thead>
          <tbody
            class=""
            role="rowgroup"
          >
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                US
              </td>
              <td
                role="cell"
              >
                18755922
              </td>
              <td
                role="cell"
              >
                330241
              </td>
              <td
                role="cell"
              >
                0
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                India
              </td>
              <td
                role="cell"
              >
                10169118
              </td>
              <td
                role="cell"
              >
                147343
              </td>
              <td
                role="cell"
              >
                9740108
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Brazil
              </td>
              <td
                role="cell"
              >
                7448560
              </td>
              <td
                role="cell"
              >
                190488
              </td>
              <td
                role="cell"
              >
                6512717
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Russia
              </td>
              <td
                role="cell"
              >
                2963290
              </td>
              <td
                role="cell"
              >
                52985
              </td>
              <td
                role="cell"
              >
                2374589
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                France
              </td>
              <td
                role="cell"
              >
                2604595
              </td>
              <td
                role="cell"
              >
                62548
              </td>
              <td
                role="cell"
              >
                195365
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                United Kingdom
              </td>
              <td
                role="cell"
              >
                2227947
              </td>
              <td
                role="cell"
              >
                70302
              </td>
              <td
                role="cell"
              >
                4822
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Turkey
              </td>
              <td
                role="cell"
              >
                2118255
              </td>
              <td
                role="cell"
              >
                19371
              </td>
              <td
                role="cell"
              >
                1970803
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Italy
              </td>
              <td
                role="cell"
              >
                2028354
              </td>
              <td
                role="cell"
              >
                71359
              </td>
              <td
                role="cell"
              >
                1377109
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Spain
              </td>
              <td
                role="cell"
              >
                1854951
              </td>
              <td
                role="cell"
              >
                49824
              </td>
              <td
                role="cell"
              >
                150376
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Germany
              </td>
              <td
                role="cell"
              >
                1632736
              </td>
              <td
                role="cell"
              >
                29580
              </td>
              <td
                role="cell"
              >
                1235968
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Colombia
              </td>
              <td
                role="cell"
              >
                1574707
              </td>
              <td
                role="cell"
              >
                41690
              </td>
              <td
                role="cell"
              >
                1430709
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Argentina
              </td>
              <td
                role="cell"
              >
                1574554
              </td>
              <td
                role="cell"
              >
                42422
              </td>
              <td
                role="cell"
              >
                1384277
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Mexico
              </td>
              <td
                role="cell"
              >
                1372243
              </td>
              <td
                role="cell"
              >
                121837
              </td>
              <td
                role="cell"
              >
                1023916
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Poland
              </td>
              <td
                role="cell"
              >
                1249079
              </td>
              <td
                role="cell"
              >
                26992
              </td>
              <td
                role="cell"
              >
                984683
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Iran
              </td>
              <td
                role="cell"
              >
                1189203
              </td>
              <td
                role="cell"
              >
                54440
              </td>
              <td
                role="cell"
              >
                933736
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Ukraine
              </td>
              <td
                role="cell"
              >
                1041583
              </td>
              <td
                role="cell"
              >
                18255
              </td>
              <td
                role="cell"
              >
                670437
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Peru
              </td>
              <td
                role="cell"
              >
                1003982
              </td>
              <td
                role="cell"
              >
                37317
              </td>
              <td
                role="cell"
              >
                940040
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                South Africa
              </td>
              <td
                role="cell"
              >
                983359
              </td>
              <td
                role="cell"
              >
                26276
              </td>
              <td
                role="cell"
              >
                830251
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Netherlands
              </td>
              <td
                role="cell"
              >
                755038
              </td>
              <td
                role="cell"
              >
                11015
              </td>
              <td
                role="cell"
              >
                8933
              </td>
            </tr>
            <tr
              role="row"
            >
              <td
                role="cell"
              >
                Indonesia
              </td>
              <td
                role="cell"
              >
                700097
              </td>
              <td
                role="cell"
              >
                20847
              </td>
              <td
                role="cell"
              >
                570304
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
